### PR TITLE
Comment System

### PR DESCRIPTION
**Pull request by TadjAdams**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,49 @@
...
+
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+        
+    response.setContentType("application/json;");
+  }
```
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Formatting is off here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,49 @@
...
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    //response.setContentType("text/html");
+
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
```
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Formatting is off for this loop

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -19,14 +19,49 @@
...
+@WebServlet("text/html")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> textBoxMessages= new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    //response.setContentType("text/html");
```
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Dead code should be deleted.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,14 +19,49 @@
...
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("user-comments", userComment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    response.sendRedirect("/index.html");
+
+
+    //response.setContentType("text-input");
```
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Dead code should be deleted.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -26,3 +26,12 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+async function getData() {
+  const response = await fetch("text/html");
+  const data = await response.text();
+  document.getElementById('data-container').innerText = data;
+}
+
```
Comment (jojojames) [inline portfolio/src/main/webapp/script.js:35]: Extra newlines are unnecessary.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
-  width: 650px;
+  width: 750px;
 }
 
 #greeting-container {
   margin-top: 20px;
 }
+body {
+    background-color: rgb(200,200,200);
+    }
```
Comment (jojojames) [inline portfolio/src/main/webapp/style.css]: Formatting is inconsistent here.

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -1,9 +1,15 @@
...
   margin-right: auto;
-  width: 650px;
+  width: 750px;
 }
 
 #greeting-container {
   margin-top: 20px;
 }
+body {
+    background-color: rgb(200,200,200);
```
Comment (jojojames) [inline portfolio/src/main/webapp/style.css:13]: 4 space indent here but this css file uses 2 space indent.

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -1,9 +1,15 @@
+
```
Comment (jojojames) [inline portfolio/src/main/webapp/style.css:1]: Extra line here is unnecessary.

File: portfolio/src/main/webapp/index.html [hunk 9]
```diff
@@ -9,10 +9,27 @@
...
-      <p>This is my portfolio.</p>
+      <p> Hello World, and all that jazz...</p>
+      <p> I'm Tadj, a rising sophomore at Howard University.  </p>
+      <p> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. </p>
+      <p> Projects  </p>
+      <a href = "https://github.com/TadjAdams"> If you want to check out my socials</a>
       <p>Click here to get a random message:</p>
       <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+        <div id="greeting-container"></div>
```
Comment (jojojames) [inline portfolio/src/main/webapp/index.html:19]: Formatting looks off here.

File: portfolio/src/main/webapp/index.html [hunk 10]
```diff
@@ -9,10 +9,27 @@
...
     </div>
+
+    <div id="content">
+        <p> Click below for a piece of sage advice: </p>
```
Comment (jojojames) [inline portfolio/src/main/webapp/index.html:31]: Formatting in this div is inconsistent with the rest of the file (it uses 4 space indent, it seems).

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 11]
```diff
@@ -19,14 +19,49 @@
...
+    String userText = request.getParameter(request, "text-input", "");
+    textBoxMessages.add(userText);
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("user-comments", userComment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    response.sendRedirect("/index.html");
+
```
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Extra new lines are unncessary.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (jojojames) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Not deleted yet.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The formatting is off.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (jojojames) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Delete dead code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (TadjAdams) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Resolved

**Pull request merged**